core/trino-main/src/main/java/io/trino/server/protocol/Query.java

```diff
@@ -545,14 +545,14 @@ private synchronized void closeExchangeClientIfNecessary(QueryInfo queryInfo)
         }
     }
 
-    private void handleSerializationException(Throwable exception)
+    private synchronized void handleSerializationException(Throwable exception)
     {
         // failQuery can throw exception if query has already finished.
         try {
             queryManager.failQuery(queryId, exception);
         }
         catch (RuntimeException e) {
-            log.debug("Could not fail query", e);
+            log.debug(e, "Could not fail query");
```

Comment:
why this change? this doesn't look right,

Reply:
Good observation. This is actually a bug fix, found by IntelliJ warnings. Our logger API takes the exception as the first argument: https://github.com/airlift/airlift/blob/5849abe7466b007ec94d4c388918983172703fbd/log/src/main/java/io/airlift/log/Logger.java#L65

Before this change, the exception is an argument to the formatted message, which is wrong since the message doesn't have a placeholder for it (and it wouldn't get logged properly as an exception): https://github.com/airlift/airlift/blob/5849abe7466b007ec94d4c388918983172703fbd/log/src/main/java/io/airlift/log/Logger.java#L93

Reply:
I guess people (at least me + the original line author) are just used to `log.info("xxxx {}", param1, exception)` logging paradigm, so I would suggest fix from where this is rooted (airlift).

Reply:
@zinking the Airlift logger API was intentionally designed to have the exception first, since that allows using it unambiguously with string formatting arguments. We can catch it with [Error Prone](https://errorprone.info/bugpattern/FormatStringAnnotation) after merging https://github.com/airlift/airlift/pull/922